x-pack/plugins/apm/server/routes/infrastructure/route.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import * as t from 'io-ts';
+import { createApmServerRoute } from '../apm_routes/create_apm_server_route';
+import { setupRequest } from '../../lib/helpers/setup_request';
+import { environmentRt, kueryRt, rangeRt } from '../default_api_types';
+import { getInfrastructureData } from './get_infrastructure_data';
+import { getMetricIndices } from '../../lib/helpers/get_metric_indices';
+import { getHostNames } from './get_host_names';
+
+const infrastructureRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/services/{serviceName}/infra',
+  params: t.type({
+    path: t.type({
+      serviceName: t.string,
+    }),
+    query: t.intersection([kueryRt, rangeRt, environmentRt]),
+  }),
+  options: { tags: ['access:apm'] },
+  handler: async (
+    resources
+  ): Promise<{
+    containerIds: string[];
+    hostNames: string[];
+    podNames: string[];
+  }> => {
+    const setup = await setupRequest(resources);
+
+    const { context, params, plugins } = resources;
+
+    const {
+      path: { serviceName },
+      query: { environment, kuery, start, end },
+    } = params;
+
+    const infrastructureData = await getInfrastructureData({
+      setup,
+      serviceName,
+      environment,
+      kuery,
+      start,
+      end,
+    });
+
+    const containerIds = infrastructureData.containerIds;
+    let containerHostNames;
```

Comment:
I'd probably export lines 53-68 to another function and change it to const:
```suggestion
    const containerHostNames = await getHostName(...)
```